### ISSUE TITLE
Finetuning Yolov3 (darknet53.conv74) on coco not working very well after 75k iterations

### ISSUE THREAD

**Issue by lihuang3** [AlexeyAB/darknet#1042]
Dear Alexey,

Thank you for providing us a great version of darknet! 

I'm trying to reproduce yolov3 coco mAP results from `darknet53.conv74` weights.

After around 60k iterations, the original code without `stopbackward =1` can have some basic detection for coco classes but poor mAP. 

So I just switched to your code (latest version in June) and set `stopbackward = 1` exactly at the line you mentioned in _"to speedup training (with decreasing detection accuracy) do Fine-Tuning instead of Transfer-Learning"_, with `yolov3.cfg` and the following changes in the config. First 5k iterations single GPU, and afterwards 2 GPUs.
```
learning_rate=0.0001
burn_in=1000
max_batches = 100000
policy=steps
steps=50000,80000
scales=.5,.2
```
After 75k iterations, I cannot detect either "dog", "bicycle" or "truck" in `data/dog.jpg`, but wrong labels at the top left corner showing "person". This seems to be weird since I have frozen the backbone weights, and it should at least have some classification results comparable to the orginal darknet code (the one I trained for 60k iterations without `stopbackward = 1`) .

So I was wondering if you could provide some insight of such results to improve the training?

Also I have a question about setting `stopbackward=1` . It is suggested to be inserted at [L598](https://github.com/AlexeyAB/darknet/blob/0039fd26786ab5f71d5af725fc18b3f521e7acfd/cfg/yolov3.cfg#L598), as you mentioned, right before the 1st yolo layer (so it is conv-79). `darknet53.conv74` only extracts layers 0-74 from the imagenet pretrained weights, which means layers 75-79 weights are frozen but they are not part of the imagenet pretrained weights. So are layers 75-79 weights randomly initialized? Does this mean I should actually freeze weights from layer 74, i.e. [L567](https://github.com/AlexeyAB/darknet/blob/0039fd26786ab5f71d5af725fc18b3f521e7acfd/cfg/yolov3.cfg#L567) instead of layer 79?

Thank you,

Li

**Comment by AlexeyAB** [comment]
@lihuang3 Hi,

> I'm trying to reproduce yolov3 coco mAP results from darknet53.conv74 weights.
> 
> After around 60k iterations, the original code without stopbackward =1 can have some basic detection for coco classes but poor mAP.

1. What mAP did you get?

> So I just switched to your code (latest version in June) and set stopbackward = 1 exactly at the line you mentioned
> ...
> After 75k iterations, I cannot detect either "dog", "bicycle" or "truck" in data/dog.jpg, but wrong labels at the top left corner showing "person".
> 

2. What mAP did you get after Fine-Tuning?
3. What command did you use for Fine-Tuning?

**Comment by lihuang3** [comment]
@AlexeyAB 
Thanks for your quick response. 
1. mAP is about 5%
2. I didn't try mAP since there isn't any correct detection on 'dog.jpg'. But if you think it is helpful to identify the problem, I can add it later.
3. I use `./darknet detector train cfg/coco.data cfg/yolov3.cfg dark53.conv74 -gpus 0,1`

And I was thinking of moving `stopbackward=1` to [L548](https://github.com/AlexeyAB/darknet/blob/0039fd26786ab5f71d5af725fc18b3f521e7acfd/cfg/yolov3.cfg#L548) where there are a couple of "#" symbols.

Thank you,

Li

**Comment by AlexeyAB** [comment]
I fixed manual: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> to speedup training (with decreasing detection accuracy) do Fine-Tuning instead of Transfer-Learning, set param stopbackward=1 here: https://github.com/AlexeyAB/darknet/blob/6d44529cf93211c319813c90e0c1adb34426abe5/cfg/yolov3.cfg#L548

----

Yes, if you use pre-trained weights `darknet53.conv.74` (74 layers) then you should set `stopbackward=1` in the shortcut-layer-73: https://github.com/AlexeyAB/darknet/blob/6d44529cf93211c319813c90e0c1adb34426abe5/cfg/yolov3.cfg#L548

----

Only if you use 81 layers that you got by this command: https://github.com/AlexeyAB/darknet/blob/6d44529cf93211c319813c90e0c1adb34426abe5/build/darknet/x64/partial.cmd#L21
or if you use whole pre-trained weights file `yolov3.weights` for training with flag `-clear`  then you can set `stopbackward=1` here: https://github.com/AlexeyAB/darknet/blob/6d44529cf93211c319813c90e0c1adb34426abe5/cfg/yolov3.cfg#L598

**Comment by lihuang3** [comment]
@AlexeyAB Thanks for explaining it so well! I appreciate it!
